lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -88,22 +88,37 @@ module.exports = {
 
   // Gives the least recent last modify date across all the functions in the service.
   async getFunctionsEarliestLastModifiedDate() {
+    let fetchError;
     const getFunctionResults = this.serverless.service.getAllFunctions().map((funName) => {
       const functionObj = this.serverless.service.getFunction(funName);
       return this.provider
         .request('Lambda', 'getFunction', {
           FunctionName: functionObj.name,
         })
         .then((res) => new Date(res.Configuration.LastModified))
-        .catch(() => new Date(0)); // Function is missing, needs to be deployed
+        .catch((err) => {
+          fetchError = err;
```

Comment:
What do you think about inspecting the error here and setting e.g. `couldNotAccessFunction` flag instead of assigning the whole error?